dts/bindings/misc/test,resources.yaml

```diff
@@ -0,0 +1,27 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+title: Resources for use in test and sample code
+
+description: >
+    This binding specifies resources that can be used in specific tests,
+    such as board pins for GPIO signals.
+
+compatible: "test,resources"
+
+properties:
+    gpios:
+        type: phandle-array
```

Comment:
DTS was created as a way to describe the hardware and its initial configuration during the boot process. So far we've kept the promise. `gpios` property is always attached to a hardware module. Even `"gpio-leds"` and `"gpio-keys"` bindings that we use to instantiate LEDs and buttons refer to real hardware. In this case we break that contract since `"test,resources"` is not a piece of hardware.

I'm personally not against introducing such a "virtual" component. It does solve neatly a [class of problems](https://github.com/zephyrproject-rtos/zephyr/pull/18544#issuecomment-524052628). However, if we do so I would rather have a solution that's more general. IMHO the #18544 does a much better job at this.

Another way of handling the issue, which I believe would be more in line with the spirit of DTS, would be to add `gpios` property directly to the `soc` node. Since that's what we actually want to do, we need to name some pins that belong to SoC and attach them to the respective `gpio-controller`.

The third option, that I see, would be to stick with what we've been doing so far: no virtual components, no `gpios` property attached to the `soc` node, instead nodes that describe a real piece of hardware. In terms of hardware resources that we need to run the tests we should have a way to describe a short and probably also a pin that is not connected.